pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1906,10 +1929,18 @@ func buildTLS(
 				return out, nil
 			}
 		}
-		if len(tls.CertificateRefs) != 1 {
+
+		val, ok := tls.Options[constants.GatewayListenerTLSOptionsIstioMeshProvidedTLS]
```

Comment:
How does this difer from ISTIO_MUTUAL?

Reply:
Ah, I missed that we already do this conversion. Will switch to gateway.istio.io/tls-terminate-mode